zenoh-flow/src/runtime/dataflow/instance/runners/connector.rs

```diff
@@ -109,7 +109,7 @@ impl Runner for ZenohSender {
         HashMap::with_capacity(0)
     }
 
-    async fn get_input_links(&self) -> HashMap<PortId, LinkReceiver<Message>> {
+    async fn take_input_links(&self) -> HashMap<PortId, LinkReceiver<Message>> {
```

Comment:
As discussed, we should transform `self.link` and make it a `Mutex<Option<…>` to actually "take" it